File: Ex4/ex4a.c
#include <stdio.h>

void fibonacci(int n) {
    int a = 0, b = 1, c;
    printf("Fibonacci Sequence: %d %d ", a, b);
    while ((a + b) <= n) {
        c = a + b;
        printf("%d ", c);
        a = b;
        b = c;
    }
}

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);
    fibonacci(num);
    return 0;
}

File: Ex4/ex4a.py
def fibonacci(n):
    a, b = 0, 1
    sequence = []
    while a <= n:
        sequence.append(a)
        a, b = b, a + b
    return sequence


def main():
    num = int(input("Enter a number: "))
    print("Fibonacci Sequence:", *fibonacci(num))

if __name__ == "__main__":
    main()
